### ISSUE TITLE
components attached by joints don't have correct orientation when exported

### ISSUE THREAD

**Issue by Robokan** [WilkoV/Fusion360_FilteredExport#7]
If you export nested components as STLs using your tool and there are components that are attached by joints so their positions and orientations have changed this is not reflected in the generated STL files. The STL files continue to have the original orientations not the correct ones.

If you use the build in export as STL it preserves these orientation changes.

**Comment by WilkoV** [comment]
Can you please provide an example file and what export settings you're using?

**Comment by Robokan** [comment]
if I export “main" which includes “attach” in the regular way using build in “save as STL files” I get this if I load it in blender:



which is correct. With your exporter I get:



it would seem that the translation and rotation made when the joint was attached is lost


Btw, I really like your exporter. The built in one is very inconsistent creating lots of duplicates and mistakes. Only deal breaker so far is this joint issue.

Thanks.

**Comment by WilkoV** [comment]
Are you having several occurrences of the same component? And are those the exports that are making problems?

**Comment by Robokan** [comment]
Yes

**Comment by Robokan** [comment]
Actually in the example I sent there is only one occurrence and the orientation is completely lost. 

Were you able to open the example? It just imports a component and then attaches it by a joint. The orientation and position change done by the joint is what is lost.

**Comment by WilkoV** [comment]
I have not forgotten this problem and have also found a solution. However, I implemented it in another add-in. Sure - two add-ins that do similar things, we don't need them unnecessarily. But Filtered export was pretty community driven and I wasn't quite happy with a few things. That's why I chose a second add-in.
Maybe you want to take a look at it. It is still at a very early stage and can currently only export STLs, but you might be interested.

https://github.com/WilkoV/Fusion360_ExportIt

**Comment by Robokan** [comment]
I finally got a chance to look at your exporter. 

How do you change the export directory? On the mac it is grayed out and won’t even let me see what directory it is exporting to. It is not editable either so I can’t change it.

**Comment by Robokan** [comment]
I did figure out where the files were exported to however when I imported the files into blender all the STL files were massive.Turns out if default units are not set in fusion360, the new design will have different units on the export.

**Comment by WilkoV** [comment]
STL files don't have a unit. Some applications do assume cm, some are
assuming inches and some are asking on import. What units do you use in
your designs?
BTW: This add-in is deprecated and replaced with
https://github.com/WilkoV/Fusion360_ExportIt. It's easier to use and has
more or less the same features if not more.

**Comment by Robokan** [comment]
Yes I am using your new Exportit. 

I realize STLs don’t have units. However when your exporter creates a new design it does not transfer over the same units as the original design. In my case my design was in meters and the design created by exporter was in millimeters. The exported STL’s ended up huge.

The other thing I found is linked components are problematic. There are file names that contain component X but the STL’s are from comment Y.

**Comment by WilkoV** [comment]
Would you mind creating an issue for ExportIt. I will check it later..

**Comment by WilkoV** [comment]
I just did a quick test:

- Create a new document
- Changed the document settings from mm to m
- Created a 1m x 1m x 1m cube
- Crated an export
- You're right that the document settings in the export helper are still on
mm (I guess because it's my default setting in the application), but the
dimensions of the cube are now 1000mm. So technically nothing changed...

If this topic is still interesting for you, please let me know by opening a
new issue here: https://github.com/WilkoV/Fusion360_ExportIt/issues

**Comment by Robokan** [comment]
Yes that’s exactly what happens. It’s not a problem unless I open it in blender expecting it to be in the units the original design was in. In that case it will be the wrong size and it took me awhile to figure out why. In my case I did not have default units set either. 

I will see if I can come up with a test case for the component issue I mentioned.

Thanks

**Comment by WilkoV** [comment]
If it's not about the default units I don't see a way to do anything about
it, because I can only copy the body to a new document. I double check it,
but I'm 90% sure that I can't.
